src/store/helpers.ts

```diff
@@ -1,4 +1,5 @@
 import type { V1ListSourceResponse } from './provisioningApi';
 
-export const extractProvisioningList = (list: V1ListSourceResponse) =>
-  list?.data;
+export const extractProvisioningList = (
+  list: V1ListSourceResponse | undefined
+) => list?.data;
```

Comment:
Why is this change needed? The undefined option seems to be included in the original API schema already.

```typescript
(alias) type V1ListSourceResponse = {
    data?: {
        id?: string | undefined;
        name?: string | undefined;
        source_type_id?: string | undefined;
        uid?: string | undefined;
    }[] | undefined;
```

Reply:
because I am using this in ReviewStepTextList- const { data: rawAzureSources, isSuccess: isSuccessAzureSources } =
    useGetSourceListQuery({ provider: 'azure' });
  const azureSources = extractProvisioningList(rawAzureSources); 
  and its complain on - 
  TS2345: Argument of type 'V1ListSourceResponse | undefined' is not assignable to parameter of type 'V1ListSourceResponse'.   Type 'undefined' is not assignable to type 'V1ListSourceResponse'